### ISSUE TITLE
Inconsistent false positive RST214

### ISSUE THREAD

**Issue by RivenSkaye** [peterjc/flake8-rst-docstrings#67]
I'm using this to make sure my rST is correct for documentation builds, but every once in a while I end up getting a weird false positive on RST214: inline literal start-string without end-string. Now it wouldn't be an issue if I could identify a cause, but I'm currently getting the warning on a piece of docstring that is _an exact duplicate_ from elsewhere in the file.

![image](https://user-images.githubusercontent.com/14993038/179837541-25413f1d-eb80-458a-9e15-416eb8797af8.png)
![image](https://user-images.githubusercontent.com/14993038/179837576-05dad0c9-63fe-4bdb-9b65-b23acb5d3162.png)

[the two functions](https://github.com/RivenSkaye/rvsfunc/blob/93b1d7b4411bb8ecc3a29f6c8bff7be4380aba6b/rvsfunc/utils.py#L99) start from this line and downwards

**Comment by peterjc** [comment]
In-line literals are using back-ticks,

https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#inline-markup

RST has some complicated rules about the boundaries:

https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#inline-markup-recognition-rules

In your case, I think it is this snippet:

```
`int`s
```

In which case there is a documented but discouraged solution:

https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#character-level-inline-markup-1

I suggest either using ints (with no markup as in the other examples), or ``int`` (singular) which to me still reads fine here.

Note if I am right, there could be an off by one error in the line number reported. That was a known issue on some combinations of Python and the plugin - could you confirm which version you are running?

**Comment by RivenSkaye** [comment]
```sh
$ pip list | grep -i rst
flake8-rst-docstrings            0.2.7
```
I don't think the off by one error is happening here. Removing the line makes the error go away entirely. I'll try removing the s and the entire `int` ref and report back tonight when I'm not at the office

**Comment by peterjc** [comment]
I wasn't clear from the screenshot if the error was reported for line 149 (wrong) or 150 (correct if it was the ints line).

**Comment by RivenSkaye** [comment]
The problem persisted. I tried the suggested removal of the s behind `int`s, betting on the off by one error, and it remained at line 149. Until I also removed the trailing s behind ``Sequence``s at 151 (this has since been pushed down to 155-157 due to formatting). So I guess the off-by-one error is back, reporting on the wrong _entry_, not just the wrong line (continuation of docs is on the next line)
![image](https://user-images.githubusercontent.com/14993038/180161823-875f6181-b24c-4df1-a499-ebd7fbcf3016.png)

**Comment by peterjc** [comment]
I'd overlooked the ``Sequence``s example after finding the ``int``s example.

Note that is using double backticks, rather than single backticks. These are different in RST (double for inline literal, single for interpreted text):

https://docutils.sourceforge.io/docs/user/rst/quickref.html#inline-markup

The error message from docutils is about inline literals, so it probably is from the ``Sequence``s example, which suggests there may indeed be an issue with the line number.

You said you're using flake8-rst-docstrings v0.2.7, but what version of Python please?

**Comment by RivenSkaye** [comment]
```sh
$ python --version
Python 3.10.5
```

**Comment by RivenSkaye** [comment]
Double backticks are very much valid rST [as per the spec](https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#inline-literals) that notes it's valid for short code samples. Sphinx also renders it as such, [docs causing the original issue](https://rvsfunc.tae.moe/en/latest/utils.html#rvsfunc.utils.replace_ranges) as an example of it being rendered as ``inline code``, much like how it works on Discord and here on github

**Comment by peterjc** [comment]
See https://github.com/peterjc/flake8-rst-docstrings/commit/ad29dfede218579020c6a90960ad72422679f098 as this limitation was not in the README.
